frontend/app/shared/components/schema-category.component.ts

```diff
@@ -110,6 +120,28 @@ export class SchemaCategoryComponent implements OnChanges {
         return schema.id;
     }
 
+    public trackByCategory(_index: number, category: SchemaCategory) {
+        return category.name;
+    }
+
+    public childSchemaCountUpdate(categoryName: string, count: number) {
+        this.childSchemaCounts[categoryName] = count;
+        this.emitSchemaCount();
+        this.cdr.detectChanges();
+    }
+
+    private emitSchemaCount() {
+        // Add up all the filtered counts of the children
```

Comment:
Yeah, looking at it again it's because the parent category count includes the schemas directly in that category and also all the schemas in child categories of that category. This means that when a category is added to or removed from a category it needs to emit a message to its parent to let it know to update its count.
There may be a better way to do this that I'm not aware of as I'm not an expert in angular.